core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestExceptionMapper.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.proxy.s3;
+
+import alluxio.exception.status.AlluxioStatusException;
+import alluxio.exception.status.FailedPreconditionException;
+import alluxio.exception.status.InvalidArgumentException;
+import alluxio.exception.status.NotFoundException;
+import alluxio.exception.status.PermissionDeniedException;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.dataformat.xml.XmlMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+import javax.ws.rs.ext.Provider;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
+/**
+ * An exception mapper for REST PROXY to convert inner exception to HTTP Response.
+ */
+@Provider
+public class S3RestExceptionMapper implements ExceptionMapper<Throwable> {
+  private static final Logger LOG = LoggerFactory.getLogger(S3RestExceptionMapper.class);
+
+  /**
+   * convert the Exception to the HTTP Response for jersey.
+   *
+   * @param e the exception to map to a response
+   * @return Response Http Response
+   */
+  @Override
+  public Response toResponse(Throwable e) {
+    if (e instanceof AlluxioStatusException) {
```

Comment:
I think currently the exception from ufs would be either AlluxioRuntimeException or IOException(we try to get rid of overuse of AlluxioStatusException). So we also need to include AlluxioRuntimeException handling here.

Reply:
OK, I'll do it. BTW I notice that the code convert other exception to AlluxioStatusException here [code link](https://github.com/Alluxio/alluxio/blob/34d46608538ed89a49c7a5ca5843bbb4c97f1d9c/core/server/worker/src/main/java/alluxio/worker/grpc/BlockReadHandler.java#L449)
And then it's why ExceptionMapper would get the AlluxioStatusExceptio.
what you mean is we try to use AlluxioRuntimeException instead of AlluxioStatusException in the future?

Reply:
yes, if the exception is not intended to be handled by the upper caller then we don't need to throw a checked exception. And a lot of time we just throw AlluxioStatusException all the way up to the client and never handle it which is not very good practice.

Reply:
Done, include AlluxioRuntimeException handling. (now the ticket is mainly about 404)